File: 프로그래머스/unrated/120878. 유한소수 판별하기/유한소수 판별하기.py
def solution(a, b):
    def gcd(a, b):
        while b:
            a, b = b, a % b
        return a
    
    g = gcd(a, b)
    a, b = a// g, b // g
    
    while b % 2 == 0:
        b = b // 2
    while b % 5 == 0:
        b = b // 5
        
    if b == 1:
        return 1
    else:
        return 2

File: 프로그래머스/0/120878. 유한소수 판별하기/유한소수 판별하기.java
import java.util.*;

class Solution {
    public int solution(int a, int b) {
        int gcdValue = gcd(a,b);
        b/=gcdValue;
        while(b%2==0){
            b /=2;
        }
        while(b%5==0){
            b/=5;
        }
        return (b==1)?1:2;
    }
    static int gcd(int a, int b){
        while(b!=0){
            int temp = a%b;
            a=b;
            b=temp;
        }
        return a;
    }
}